catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -337,26 +346,28 @@ private static boolean anyNull(Object... args) {
     };
   }
 
-  /**
-   * Caution should be used with this, as it elevates the permissions to the System user.
-   *
-   * <p>Operations that throw an {@link Exception} will be tried (in total) a maximum of 3 times
-   * before actually failing.
-   *
-   * @param func What to execute as the System
-   * @param <T> Generic return type of func
-   * @return result of the callable func
-   */
-  private static <T> T executeAsSystem(Callable<T> func) {
-    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
-    if (systemSubject == null) {
-      throw new SecurityException("Could not get systemSubject to initialize system templates");
+  private static void createSystemMetacards(
+      EndpointUtil util, List<Metacard> systemTemplates, CatalogFramework catalogFramework) {
+    Set<String> queryTitles = titlesTransform(util.getMetacardsByFilter(QUERY_TEMPLATE_TAG));
+    Set<String> resultTitles = titlesTransform(util.getMetacardsByFilter(ATTRIBUTE_GROUP_TAG));
+    List<Metacard> dedupedTemplateMetacards =
+        Stream.concat(
+                systemTemplates
+                    .stream()
+                    .filter(QueryTemplateMetacard::isQueryTemplateMetacard)
+                    .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
+                systemTemplates
+                    .stream()
+                    .filter(AttributeGroupMetacard::isAttributeGroupMetacard)
+                    .filter(metacard -> !resultTitles.contains(metacard.getTitle())))
+            .collect(Collectors.toList());
+
+    if (!dedupedTemplateMetacards.isEmpty()) {
+      try {
+        catalogFramework.create(new CreateRequestImpl(dedupedTemplateMetacards));
```

Comment:
:question: Which of these two calls would actually throw the exceptions being caught?

Reply:
`catalogFramework.create`